CMakeLists.txt

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2019 Linaro
+#
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.13.1)
+
+# Set default image versions if not defined elsewhere
+if (NOT DEFINED TFM_IMAGE_VERSION_S)
+	set(TFM_IMAGE_VERSION_S 0.0.0+0)
+endif()
+
+if (NOT DEFINED TFM_IMAGE_VERSION_NS)
+	set(TFM_IMAGE_VERSION_NS 0.0.0+0)
+endif()
+
+function(trusted_firmware_build)
+  set(oneValueArgs BINARY_DIR BOARD OUT_VENEERS_FILE OUT_INCLUDE_PATH)
+  cmake_parse_arguments(TFM "" "${oneValueArgs}" "" ${ARGN})
+
+  set(TFM_REMOTE_DIR "${ZEPHYR_TFM_MODULE_DIR}")
+  set(TFM_BASE_DIR "${TFM_REMOTE_DIR}/trusted-firmware-m")
+
+  set(VENEERS_FILE ${TFM_BINARY_DIR}/install/export/tfm/veneers/s_veneers.o)
+  set(${TFM_OUT_VENEERS_FILE} ${VENEERS_FILE} PARENT_SCOPE)
+  set(${TFM_OUT_INCLUDE_PATH} ${TFM_BINARY_DIR}/install/export/tfm/include PARENT_SCOPE)
+
+  # Could be argument.
+  set(TFMCONFIG "ConfigRegressionIPC") 
+
+  include(ExternalProject)
+  
+  ExternalProject_Add(
+    tfm
+    SOURCE_DIR ${ZEPHYR_TFM_MODULE_DIR}/trusted-firmware-m
+    BINARY_DIR ${TFM_BINARY_DIR}
+    BUILD_BYPRODUCTS ${VENEERS_FILE}
+    CMAKE_ARGS -DPROJ_CONFIG=${TFM_BASE_DIR}/configs/${TFMCONFIG}.cmake
+               -DTARGET_PLATFORM=${TFM_BOARD} -DBL2=True -DCOMPILER=GNUARM
```

Comment:
Can these values come from our Zephyr toolchain params are set.  ie based ZEPHYR_TOOLCHAIN_VARIANT and such?

That way if one is using the ARM embedded toolchain things would work.

Reply:
I'll leave that to @microbuilder to look into.
I just created this draft to help him get going with moving the code from  `samples/tfm_integration/tfm_ipc/CMakeLists.txt` into the Zephyr module, so that he has a starting point for continues work.

I made no consideration regarding cleanup.
But I agree with @galak there is more work to do, than just moving the code itself.

Reply:
@galak Something like this?

```
  # Check the toolchain variant
  if(${ZEPHYR_TOOLCHAIN_VARIANT} STREQUAL "zephyr")
    set(TFM_TOOLCHAIN "GNUARM")
    set(TFM_TOOLCHAIN_PREFIX "arm-zephyr-eabi")
    set(TFM_TOOLCHAIN_PATH ${ZEPHYR_SDK_INSTALL_DIR})
  elseif(${ZEPHYR_TOOLCHAIN_VARIANT} STREQUAL "gnuarmemb")
    set(TFM_TOOLCHAIN "GNUARM")
    set(TFM_TOOLCHAIN_PREFIX "arm-none-eabi")
    set(TFM_TOOLCHAIN_PATH ${GNUARMEMB_TOOLCHAIN_PATH})
  else()
    message(FATAL_ERROR "ZEPHYR_TOOLCHAIN_VARIANT must be zephyr or gnuarmemb!")
  endif()

  include(ExternalProject)

  ExternalProject_Add(
    tfm
    SOURCE_DIR ${TFM_BASE_DIR}
    BINARY_DIR ${TFM_BINARY_DIR}
    BUILD_BYPRODUCTS ${VENEERS_FILE}
    CMAKE_ARGS -DPROJ_CONFIG=${TFM_BASE_DIR}/configs/${TFMCONFIG}.cmake
               -DTARGET_PLATFORM=${TFM_BOARD}
               -DBL2=True
               -DCOMPILER=${TFM_TOOLCHAIN}
               -DGNUARM_PREFIX=${TFM_TOLCHAIN_PREFIX}
               -DGNUARM_PATH=${ZEPHYR_SDK_INSTALL_DIR}
  )
```

Reply:
it's not enough to just error out.
There are more toolchains that could be valid.

For example, I would assume this to work as well:
```
# Linux, macOS:
$ echo $ZEPHYR_TOOLCHAIN_VARIANT
cross-compile
$ echo $CROSS_COMPILE
/usr/bin/arm-none-eabi-
```
https://docs.zephyrproject.org/latest/getting_started/toolchain_other_x_compilers.html#other-cross-compilers

Reply:
TF-M has very specific requirements for the toolchain used. Passing in an unknown cross-compiler here may cause as many problems as it solves? I can add this in and pass it, it's just an unknown if it will actually work or not, and TF-M has checks for specific versions of GCC, etc.

Reply:
well, in this case, a default ubuntu / debian / Redhat / Fedore gcc-arm toolchain is probably a more known toolchain to TF-M than the Zephyr SDK toolchain.
So if Zephyr SDK can be used, then I would assume a distro version should also.

From the link I posted above.
```
# On Debian or Ubuntu
sudo apt-get install gcc-arm-none-eabi
# On Fedora or Red Hat
sudo dnf install arm-none-eabi-newlib
```